### ISSUE TITLE
After training for time, the loss becomes 0

### ISSUE THREAD

**Issue by Lebesgue-zyker** [cwmok/C2FViT#11]
I used the linked data you provided to train Train_C2FViT_pairwise.py without modifying the code. At the beginning, loss was negative, but sometimes it would become positive and finally become 0. Do you know what happened?
![image](https://github.com/cwmok/C2FViT/assets/54661026/fcc938f7-6f80-42b2-8f79-3bb69a6f4c01)

**Comment by Lebesgue-zyker** [comment]
It looks like the problem is with the loss function, which I replaced with MAE to train correctly. But why does the model converge so slowly, is it a problem with MSE? I have trained 80,000 iteration, and I have not achieved the effect of 18,000 iterationof pre-training models.

**Comment by cwmok** [comment]
Hi @Lebesgue-zyker,

My model works fine with NCC loss. However, I noticed that some users also reported the inability issue with NCC loss, as mentioned in https://github.com/Project-MONAI/MONAI/discussions/3463. 

In my own experiments, I also found that using NCC will converge faster and better than MAE/MSE loss. 

Perhaps, you should try another implementation of NCC such as the one in https://github.com/Project-MONAI/MONAI/blob/8a70678baa976a01274f55837e339f2c2975dec7/monai/losses/image_dissimilarity.py#L51

**Comment by wydilearn** [comment]
Hi @cwmok, I have tried all the solutions you provided. But I still can't deal with the problem well. Can you provide more better suggestions or codes?
Thank you for your project and I'm looking forward to receiving your reply.

**Comment by wydilearn** [comment]
> Hi @Lebesgue-zyker,
> 
> My model works fine with NCC loss. However, I noticed that some users also reported the instability issue with NCC loss, as mentioned in [Project-MONAI/MONAI#3463](https://github.com/Project-MONAI/MONAI/discussions/3463).
> 
> In my own experiments, I also found that using NCC will converge faster and better than MAE/MSE loss.
> 
> Perhaps, you should try another implementation of NCC such as the one in https://github.com/Project-MONAI/MONAI/blob/8a70678baa976a01274f55837e339f2c2975dec7/monai/losses/image_dissimilarity.py#L51 or the one in https://github.com/uncbiag/ICON/blob/66ad08456d6676310b61a1120f21c4626fef1604/src/icon_registration/losses.py#L578. They both works fine in my environment.

Hi @cwmok, I have tried all the solutions you provided. But I still can't deal with the problem well. Can you provide more better suggestions or codes?
Thank you for your project and I'm looking forward to receiving your reply.
![屏幕截图 2023-11-02 194917](https://github.com/cwmok/C2FViT/assets/104293103/d8ca1ac4-ab85-4419-adce-222b658f1fd6)

**Comment by cwmok** [comment]
Hi @wydilearn and @Lebesgue-zyker,

I finally figured out the problem. The problem was caused by a mistake when I tried to update the NCC loss a few months ago. I mistakenly commented out part of the code, as shown below:
![image](https://github.com/cwmok/C2FViT/assets/63316551/2bbc28cc-ed3b-41e9-8d97-93328838aa7a)

I have updated the code. The loss function should work from now on. Sorry for the mistake, and thanks for reporting this issue.
![image](https://github.com/cwmok/C2FViT/assets/63316551/19332f40-9f5b-4b2c-ab63-a61cceb1f05d)
